be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -456,6 +482,43 @@ Status LakePersistentIndex::commit(MetaFileBuilder* builder) {
     return Status::OK();
 }
 
+// rebuild index from del files.
+Status LakePersistentIndex::load_dels(const RowsetPtr& rowset, const Schema& pkey_schema, int64_t rowset_version) {
+    TRACE_COUNTER_INCREMENT("rebuild_index_del_cnt", 1);
+    // Build pk column struct from schema
+    std::unique_ptr<Column> pk_column;
+    RETURN_IF_ERROR(PrimaryKeyEncoder::create_column(pkey_schema, &pk_column));
+    // Iterate all del files and insert into index.
+    for (const auto& del : rowset->metadata().del_files()) {
+        ASSIGN_OR_RETURN(auto read_file, fs::new_random_access_file(_tablet_mgr->del_location(_tablet_id, del)));
+        ASSIGN_OR_RETURN(auto file_size, read_file->get_size());
+        std::vector<uint8_t> read_buffer(file_size);
+        // read from del file
+        RETURN_IF_ERROR(read_file->read_at_fully(0, read_buffer.data(), read_buffer.size()));
+        // serialize to column
+        auto pkc = pk_column->clone();
+        if (serde::ColumnArraySerde::deserialize(read_buffer.data(), pkc.get()) == nullptr) {
+            return Status::InternalError("column deserialization failed");
```

Comment:
Can you get more information on why it fails?

Reply:
Done

Reply:
is it appropriate to attach the filename (not the fullpath) to the error message, or make a WARN error with the given file path, so that if needs to trouble shoot the failure, it would be easier to know which file to analyze.